process/src/services/service.ports.ts

```diff
@@ -1,295 +1,406 @@
-                return reject(new Error(this._logger.error(`Fail to get port controller because options isn't ann object`)));
-            }
-            if (typeof options.path !== 'string' || options.path.trim() === '') {
-                return reject(new Error(this._logger.error(`Fail to get port controller because "path" is incorrect: ${options.path}`)));
-            }
-            let controller: ControllerSerialPort | undefined = this._controllers.get(options.path);
-            if (controller !== undefined) {
-                this._refPortListeners(session, options.path, listeners).then(resolve);
-                return;
-            }
-            controller = new ControllerSerialPort(options);
-            controller.open().then(() => {
-                if (controller === undefined) {
-                    return;
-                }
-                controller.on(ControllerSerialPort.Events.data, this._onData.bind(this, options.path));
-                controller.on(ControllerSerialPort.Events.error, this._onError.bind(this, options.path));
-                controller.on(ControllerSerialPort.Events.disconnect, this._onDisconnect.bind(this, options.path));
-                this._controllers.set(options.path, controller);
-                this._refPortListeners(session, options.path, listeners).then(resolve);                
-            }).catch((error: Error) => {
-                reject(new Error(this._logger.error(`Fail to create port's controller due error: ${error.message}`)));
-            });
+  public refPort(
+    session: string,
+    options: IOptions,
+    listeners: IListeners
+  ): Promise<void> {
+    return new Promise((resolve, reject) => {
+      if (typeof options !== "object" || options === null) {
+        return reject(
+          new Error(
+            this._logger.error(
+              `Fail to get port controller because options isn't ann object`
+            )
+          )
+        );
+      }
+      if (typeof options.path !== "string" || options.path.trim() === "") {
+        return reject(
+          new Error(
+            this._logger.error(
+              `Fail to get port controller because "path" is incorrect: ${options.path}`
+            )
+          )
+        );
+      }
+      let controller: ControllerSerialPort | undefined = this._controllers.get(
+        options.path
+      );
+      if (controller !== undefined) {
+        this._refPortListeners(session, options.path, listeners)
+          .then(resolve)
+          .catch((error: Error) => {
+            return reject(
+              new Error(
+                this._logger.error(
+                  `Fail to get port controller due to error: ${error.message}`
+                )
+              )
+            );
+          });
+        return;
+      }
+      controller = new ControllerSerialPort(options);
+      controller
+        .open()
+        .then(() => {
+          if (controller === undefined) {
+            return;
```

Comment:
Looks like in this case, a root promise never would be resolved/rejected
Probably here should be `reject`?